doc/user/tutorial.rst

```diff
@@ -350,35 +339,20 @@ Go ahead and edit your ``images.py`` file to look something like this:
 .. code:: python
 
     import os
-    import time
     import uuid
+    import mimetypes
 
     import falcon
 
 
-    def _media_type_to_ext(media_type):
-        # Strip off the 'image/' prefix
-        return media_type[6:]
-
-
-    def _ext_to_media_type(ext):
-        return 'image/' + ext
-
-
-    def _generate_id():
-        return str(uuid.uuid4())
-
-
     class Collection(object):
 
         def __init__(self, storage_path):
             self.storage_path = storage_path
 
         def on_post(self, req, resp):
-            image_id = _generate_id()
-            ext = _media_type_to_ext(req.content_type)
-            filename = image_id + '.' + ext
-
+            ext = mimetypes.guess_extension(req.content_type)
+            filename = '{uuid}{ext}'.format(uuid=uuid.uuid4(), ext=ext)
```

Comment:
`guess_extension` always [includes the leading dot](https://docs.python.org/2/library/mimetypes.html#mimetypes.guess_extension).